packages/graph/hash_graph/lib/graph/src/shared/identifier/time/projection.rs

```diff
@@ -145,104 +151,102 @@ impl<K, I> UnresolvedProjection<K, I> {
-                "kernel",
-                openapi::Ref::from_schema_name("UnresolvedDecisionTimeKernel"),
+                "pinned",
+                openapi::Ref::from_schema_name("UnresolvedDecisionTimestamp"),
             )
-            .required("kernel")
+            .required("pinned")
             .property(
-                "image",
-                openapi::Ref::from_schema_name("UnresolvedTransactionTimeImage"),
+                "variable",
+                openapi::Ref::from_schema_name("UnresolvedTransactionTimeInterval"),
             )
-            .required("image")
+            .required("variable")
             .into()
     }
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(untagged)]
-pub enum UnresolvedTimeProjection {
-    DecisionTime(UnresolvedProjection<TransactionTime, DecisionTime>),
-    TransactionTime(UnresolvedProjection<DecisionTime, TransactionTime>),
+pub enum UnresolvedTimeAxes {
+    DecisionTime(UnresolvedConcreteTimeAxes<TransactionTimeAxis, DecisionTimeAxis>),
+    TransactionTime(UnresolvedConcreteTimeAxes<DecisionTimeAxis, TransactionTimeAxis>),
 }
 
-impl Default for UnresolvedTimeProjection {
+impl Default for UnresolvedTimeAxes {
     fn default() -> Self {
-        Self::DecisionTime(UnresolvedProjection {
-            kernel: UnresolvedKernel::new(None),
-            image: UnresolvedImage::new(
+        Self::DecisionTime(UnresolvedConcreteTimeAxes {
+            pinned: UnresolvedTaggedTimestamp::new(None),
+            variable: UnresolvedTaggedTimeInterval::new(
                 Some(TimeIntervalBound::Unbounded),
                 Some(TimeIntervalBound::Unbounded),
             ),
         })
     }
 }
 
-impl UnresolvedTimeProjection {
+impl UnresolvedTimeAxes {
     #[must_use]
-    pub fn resolve(self) -> TimeProjection {
+    pub fn resolve(self) -> TimeAxes {
         match self {
-            Self::DecisionTime(projection) => TimeProjection::DecisionTime(projection.resolve()),
-            Self::TransactionTime(projection) => {
-                TimeProjection::TransactionTime(projection.resolve())
-            }
+            Self::DecisionTime(axes) => TimeAxes::DecisionTime(axes.resolve()),
+            Self::TransactionTime(axes) => TimeAxes::TransactionTime(axes.resolve()),
         }
     }
 }
 
-impl ToSchema for UnresolvedTimeProjection {
+impl ToSchema for UnresolvedTimeAxes {
     fn schema() -> openapi::RefOr<openapi::Schema> {
         openapi::OneOfBuilder::new()
+            .item(openapi::Ref::from_schema_name("UnresolvedDecisionTimeAxes"))
             .item(openapi::Ref::from_schema_name(
-                "UnresolvedDecisionTimeProjection",
-            ))
-            .item(openapi::Ref::from_schema_name(
-                "UnresolvedTransactionTimeProjection",
+                "UnresolvedTransactionTimeAxes",
             ))
             .into()
     }
 }
 
-/// The pinned axis of a [`TimeProjection`].
+/// The pinned axis of a [`TimeAxes`].
```

Comment:
Can we improve this documentation? It's an axis pinned at a given `Timestamp` or something? (just to explain a little what the variables are, I'm familiar with this and it still took me a few seconds)

Reply:
Also if this is _**only**_ used for a pinned axis, why call it `TaggedTimestamp` instead of `PinnedAxis`